### PR TITLE
4844: Refactor tests to use Spec and SpecHelper classes

### PR DESCRIPTION
**Pull request by danceratopz**
This PR refactors:
- Spec constants and helper functions to their own `Spec` class,
- Closely related constants and helper functions to a `SpecHelpers` class,
- Updates the doc script to add the `specs.py` to the online doc.
- Deletes duplicate helper module `utils.py`.

I diff'd the fixtures before and after to verify no errors crept in :heavy_check_mark:

Doc preview:
- https://danceratopz.github.io/execution-spec-tests/v1.0.1b1-refactor-4844-specs/tests/cancun/eip4844_blobs/spec/

### CODE REVIEW

**Review by marioevz — APPROVED** [review]
Awesome! Looks great, and removes a bunch of lines which is always good :)

**Comment by danceratopz** [comment]
Rebased due to changes from:
- #171
- #195

**Comment by danceratopz** [comment]
Would you like to add anything or any other ideas @spencer-tb? :slightly_smiling_face:

File: tests/cancun/eip4844_blobs/conftest.py [hunk 1]
```diff
@@ -12,16 +12,11 @@
...
-    TARGET_DATA_GAS_PER_BLOCK,
-    calc_excess_data_gas,
-    get_data_gasprice,
-)
+from .spec import BlockHeaderDataGasFields, Spec
 
 
 @pytest.fixture
-def block_intermediate(
+def block_intermediate(  # TODO: Name this fixture more clearly given scope
```
Comment (spencer-tb) [inline tests/cancun/eip4844_blobs/conftest.py:19]: Maybe something like: `dgu_from_genesis_block` or `dgu_workaround_block`

**Review by spencer-tb — APPROVED** [review]
Really like the spec file here! I'll make sure to do the same for the 4788 tests :100:

File: tests/cancun/eip4844_blobs/conftest.py (hunk 1 above)
Comment (danceratopz) [reply tests/cancun/eip4844_blobs/conftest.py:19]: As discussed: `non_zero_data_gas_used_genesis_block` :-)

**Pull request merged**